endpoints/openrtb2/auction.go

```diff
@@ -336,111 +336,136 @@ func parseTimeout(requestJson []byte, defaultTimeout time.Duration) time.Duratio
 // Preference is given to parameters at imp[].ext level over req.ext level.
 // Parameters at req.ext level are propagated to adapters as is without any validation.
 func mergeBidderParams(req *openrtb_ext.RequestWrapper) error {
-	reqBidderParams, err := adapters.ExtractReqExtBidderParams(req.BidRequest)
+	reqBidderParams, err := adapters.ExtractReqExtBidderParamsEmbeddedMap(req.BidRequest)
 	if err != nil {
-		return err
+		return nil
+	}
+
+	if len(reqBidderParams) == 0 {
+		return nil
 	}
```

Comment:
Optimization to avoid the expensive operations of the loop if there are no bidderparams to apply.

Reply:
I have no preference. I can combine them if you prefer.

Reply:
that's fine. No need to hold this PR over this.